pco_java/src/main/rust/src/lib.rs

```diff
@@ -68,24 +70,11 @@ fn simple_compress_inner(
 
 fn decompress_chunks<T: Number + JavaConversions>(
   env: &mut JNIEnv,
-  mut src: &[u8],
+  src: &[u8],
   file_decompressor: FileDecompressor,
 ) -> Result<jobject> {
-  let n_hint = file_decompressor.n_hint();
-  let mut res: Vec<T> = Vec::with_capacity(n_hint);
-  while let MaybeChunkDecompressor::Some(mut chunk_decompressor) =
-    file_decompressor.chunk_decompressor::<T, &[u8]>(src)?
-  {
-    let initial_len = res.len(); // probably always zero to start, since we just created res
-    let remaining = chunk_decompressor.n();
-    unsafe {
-      res.set_len(initial_len + remaining);
-    }
-    let progress = chunk_decompressor.decompress(&mut res[initial_len..])?;
-    assert!(progress.finished);
-    src = chunk_decompressor.into_src();
-  }
-  let num_array = num_array::to_java(env, &res)?;
```

Comment:
I see that this piece is a diverged almost-duplicate of the logic in `simple_decompress`, with this copy not having the bounds check.  I'll return to this in a bit to try to understand in detail what else exactly has been deleted here.